In [1]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


In [2]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import warnings

from src.features import build_features

warnings.filterwarnings("ignore")
sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_DEL_TORAX = 112103

## 1. Análisis de Procedimientos a Nivel País

En este análisis se quieren responder las siguiente preguntas:

1. ¿Cómo es el ranking de los hospitales para cada uno de los procedimientos realizads?
2. ¿Quién es el Hospital que realiza más de X procedimientos?
3. ¿El procedimiento X es realizado en todo el país? ¿En qué partes en específico se realiza?

Para responder esta pregunta se utilizará la base de datos de Egresos de Hospitales adscritos
a GRD entre 2019 y 2021.


In [3]:
df_procesada = pd.read_csv("../data/processed/df_procesada.csv", sep=";")
df_procesada["ANIO_EGRESO"] = df_procesada["ANIO_EGRESO"].astype("Int32")

In [4]:
display(df_procesada.head(10))

COD_HOSPITAL  CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO  ETNIA   PROVINCIA  \
0        118100       1314867.0   MUJER       1988-06-19  OTRO   CONCEPCION   
1        118100       1354418.0   MUJER       1988-03-27  OTRO   CONCEPCION   
2        118100        239861.0   MUJER       2017-06-17  OTRO   CONCEPCION   
3        118100        330326.0  HOMBRE       2017-12-04  OTRO   CONCEPCION   
4        118100       1369293.0   MUJER       1995-12-06  OTRO   CONCEPCION   
5        118100        469385.0   MUJER       1942-06-29  OTRO   CONCEPCION   
6        118100       1146635.0   MUJER       1951-07-14  OTRO   CONCEPCION   
7        118100         93557.0  HOMBRE       1938-03-04  OTRO   CONCEPCION   
8        118100        674943.0   MUJER       1962-08-21  OTRO   CONCEPCION   
9        118100       1079955.0  HOMBRE       1944-12-07  OTRO   CONCEPCION   

                COMUNA NACIONALIDAD                       PREVISION  \
0  SAN PEDRO DE LA PAZ        CHILE  FONASA INSTITUCIONAL - (MAI) A   
1  SAN PEDRO DE LA PAZ        CHILE  FONASA INSTITUCIONAL - (MAI) D   
2           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   
3           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) A   
4              FLORIDA        CHILE  FONASA INSTITUCIONAL - (MAI) C   
5          CHIGUAYANTE        CHILE  FONASA INSTITUCIONAL - (MAI) B   
6           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   
7                 LOTA        CHILE  FONASA INSTITUCIONAL - (MAI) D   
8          SANTA JUANA        CHILE  FONASA INSTITUCIONAL - (MAI) B   
9           CONCEPCIÓN        CHILE  FONASA INSTITUCIONAL - (MAI) B   

  SERVICIO_SALUD  ... USOSPABELLON IR_29301_COD_GRD IR_29301_PESO  \
0     CONCEPCIÓN  ...          1.0         146102.0        0.5744   
1     CONCEPCIÓN  ...          NaN         134161.0        0.2951   
2     CONCEPCIÓN  ...          NaN          64182.0        0.6736   
3     CONCEPCIÓN  ...          NaN          64172.0        0.5475   
4     CONCEPCIÓN  ...          NaN         134141.0        0.3107   
5     CONCEPCIÓN  ...          NaN         114103.0        1.1799   
6     CONCEPCIÓN  ...          NaN          44163.0        1.0328   
7     CONCEPCIÓN  ...          1.0          11101.0        1.8731   
8     CONCEPCIÓN  ...          1.0          61402.0        1.3078   
9     CONCEPCIÓN  ...          NaN          14133.0        1.6687   

  IR_29301_SEVERIDAD IR_29301_MORTALIDAD  \
0                2.0                 1.0   
1                1.0                 1.0   
2                2.0                 2.0   
3                2.0                 2.0   
4                1.0                 1.0   
5                3.0                 3.0   
6                3.0                 3.0   
7                1.0                 1.0   
8                2.0                 2.0   
9                3.0                 3.0   

                          HOSPPROCEDENCIA ESTANCIA ANIO_EGRESO MES_EGRESO  \
0                                     NaN      5.0        2019        3.0   
1                                     NaN      4.0        2019        3.0   
2                                     NaN      6.0        2019        3.0   
3                                     NaN      2.0        2019        3.0   
4                                     NaN      6.0        2019        3.0   
5                                     NaN     47.0        2019        3.0   
6                                     NaN     43.0        2019        3.0   
7                                     NaN     41.0        2019        3.0   
8  HOSPITAL CLORINDA AVELLO (SANTA JUANA)     33.0        2019        3.0   
9                                     NaN     31.0        2019        3.0   

    FECHA  
0  2019-3  
1  2019-3  
2  2019-3  
3  2019-3  
4  2019-3  
5  2019-3  
6  2019-3  
7  2019-3  
8  2019-3  
9  2019-3  

[10 rows x 133 columns]

Los procedimientos en esta base de datos se encuentran en las columnas "PROCEDIMIENTOSX", donde X
va desde 1 hasta 30. Para analizar esta base de forma más expedita, se transformará de formato
ancho a formato largo, utilizando como identificador a las columnas ["COD_HOSPITAL", "ANIO_EGRESO",
"CIP_ENCRIPTADO", "DIAGNOSTICO1"]. Cabe destacar que la columna "CIP_ENCRIPTADO" hace referencia al
RUT del paciente.


In [5]:
procedimientos_formato_long = pd.melt(
    df_procesada,
    id_vars=["COD_HOSPITAL", "ANIO_EGRESO", "CIP_ENCRIPTADO", "DIAGNOSTICO1"],
    value_vars=[f"PROCEDIMIENTO{i}" for i in range(1, 31)],
    value_name="procedimiento",
)

In [6]:
conteo_y_ranking_procedimientos = build_features.contar_procedimientos_y_rankear(
    procedimientos_formato_long
)

In [7]:
display(conteo_y_ranking_procedimientos.head(20))

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo  ranking  \
150206         2021        108100         99.99       2        1   
165765         2021        113180         99.99       2        2   
140654         2021        103100         99.99       1        3   
143682         2021        105100         99.99       1        4   
155556         2021        110120         99.99       1        5   
167921         2021        114101         99.99       1        6   
183553         2021        121109         99.99       1        7   
186513         2021        122100         99.99       1        8   
190822         2021        126100         99.99       1        9   
191999         2021        129106         99.98      19        1   
151485         2021        109100         99.98      11        2   
186788         2021        123100         99.98      11        3   
165565         2021        113180         99.98       8        4   
178447         2021        118100         99.98       8        5   
181258         2021        120101         99.98       8        6   
153822         2021        110100         99.98       7        7   
176484         2021        117101         99.98       6        8   
182651         2021        121109         99.98       6        9   
154968         2021        110120         99.98       5       10   
156780         2021        111100         99.98       5       11   

        diferencia_proximo_hospital  total_procedimientos  
150206                            0                    11  
165765                            1                    11  
140654                            0                    11  
143682                            0                    11  
155556                            0                    11  
167921                            0                    11  
183553                            0                    11  
186513                            0                    11  
190822                          -18                    11  
191999                            8                   119  
151485                            0                   119  
186788                            3                   119  
165565                            0                   119  
178447                            0                   119  
181258                            1                   119  
153822                            1                   119  
176484                            0                   119  
182651                            1                   119  
154968                            0                   119  
156780                            0                   119

El resultado muestra el ranking para todos los procedimientos asignados en el país en cada uno de 
los años analizados (2019 a 2021). 

A modo de ejemplo, se ilustra el ranking en el año 2021 para el procedimiento 99.99. En este caso,
se observa que los hospitales con código 108100 y 113180 están en los primeros lugares, realizando
2 procedimientos cada uno.

Ahora, si se filtra por el hospital del tórax, y sólo los procedimientos donde el hospital sea
ranking 1, entonces se obtendrán procedimientos altamente relevantes para el país.

In [12]:
mejores_procedimientos_torax = conteo_y_ranking_procedimientos.query(
    "COD_HOSPITAL == @HOSPITAL_DEL_TORAX and ranking == 1"
).sort_values(["ANIO_EGRESO", "conteo"],ascending=False)

In [13]:
display(mejores_procedimientos_torax)

ANIO_EGRESO  COD_HOSPITAL procedimiento  conteo  ranking  \
162203         2021        112103         39.61     312        1   
162205         2021        112103         39.64     294        1   
162211         2021        112103         42.23     189        1   
162216         2021        112103         37.22     138        1   
162221         2021        112103         88.53     129        1   
...             ...           ...           ...     ...      ...   
31575          2019        112103         39.64       1        1   
31548          2019        112103         91.61       1        1   
31486          2019        112103         39.55       1        1   
31476          2019        112103         38.46       1        1   
31474          2019        112103         38.42       1        1   

        diferencia_proximo_hospital  total_procedimientos  
162203                           37                  1582  
162205                          135                  1018  
162211                          169                   304  
162216                           10                   378  
162221                           11                   799  
...                             ...                   ...  
31575                             0                     1  
31548                             0                     3  
31486                             0                     1  
31476                             0                     3  
31474                             0                     4  

[264 rows x 7 columns]

Los resultados muestran que el tórax en el año 2021 fue altamente relevante en los procedimientos
39.61, 39.64, 42.23, entre otros. Además, se puede observar la diferencia que tiene el Tórax
con los siguientes establecimientos de salud y el total de procedimientos realizados en el país.